src/grammar/query.rs

```diff
@@ -36,6 +36,204 @@ pub(crate) fn parse_query(p: &mut Parser, expect_into_clause: bool) {
     p.finish();
 }
 
+pub(crate) fn parse_cte(p: &mut Parser) {
+    p.start(SyntaxKind::WithClause);
+    p.expect(T![with]);
+    if p.at(T![create]) {
+        parse_plsql_declarations(p);
+    }
+    safe_loop!(p, {
+        if p.nth(1) == Some(T![analytic]) {
+            parse_subav_factoring_clause(p);
+        } else if p.at(T![select]) {
+            break;
+        } else {
+            parse_subquery_factoring_clause(p);
+        }
+
+        if !p.eat(T![,]) {
+            break;
+        }
+    });
+    p.finish();
+    parse_query(p, false);
+}
+
+pub(crate) fn parse_subav_factoring_clause(p: &mut Parser) {
+    p.start(SyntaxKind::SubavFactoringClause);
+    parse_ident(p, 1..1);
+    p.expect(T![analytic]);
+    p.expect(T![view]);
+    p.expect(T![as]);
+    p.expect(T!["("]);
+    parse_subav_clause(p);
+    p.expect(T![")"]);
+    p.finish();
+}
+
+pub(crate) fn parse_subav_clause(p: &mut Parser) {
+    p.start(SyntaxKind::SubavClause);
+    p.expect(T![using]);
+    parse_ident(p, 1..2);
+    if p.at(T![hierarchies]) {
+        parse_hierarchies_clause(p);
+    }
+    if p.at(T![filter]) {
+        parse_filter_clauses(p);
+    }
+    if p.at(T![add]) {
+        parse_add_calcs_clause(p);
+    }
+    p.finish();
+}
+
+pub(crate) fn parse_hierarchies_clause(p: &mut Parser) {
+    p.start(SyntaxKind::HierarchiesClause);
+    p.expect(T![hierarchies]);
+    p.expect(T!["("]);
+    safe_loop!(p, {
+        parse_ident(p, 1..2);
+        if !p.eat(T![,]) {
+            break;
+        }
+    });
+    p.expect(T![")"]);
+    p.finish();
+}
+
+pub(crate) fn parse_filter_clauses(p: &mut Parser) {
+    p.start(SyntaxKind::FilterClauses);
+    p.expect(T![filter]);
+    p.expect(T![fact]);
+    p.expect(T!["("]);
+    safe_loop!(p, {
+        parse_filter_clause(p);
+        if !p.eat(T![,]) {
+            break;
+        }
+    });
+    p.expect(T![")"]);
+    p.finish();
+}
+
+pub(crate) fn parse_filter_clause(p: &mut Parser) {
+    p.start(SyntaxKind::FilterClause);
+    if !p.eat(T![measures]) {
+        parse_ident(p, 1..2);
+    }
+    p.expect(T![to]);
+    parse_expr(p);
+    p.finish();
+}
+
+pub(crate) fn parse_add_calcs_clause(p: &mut Parser) {
+    p.start(SyntaxKind::AddCalcsClause);
```

Comment:
We're otherwise following Oracles names for these nodes, why not `add_meas_clause` in this case?

Reply:
![grafik](https://github.com/user-attachments/assets/c5d6a9a6-080a-4394-89b9-4865a8e3332a)
this is what it's called in the docs